README.md

```diff
@@ -36,78 +39,75 @@ npm i @liftric/auth@<version>
 
 ## How-to
 
-### Instantiating
-
-The handler needs a configuration object consisting of the region code and the client ID.
-
-```kotlin
-val configuration = Configuration(region = Region.euCentral1,
-                                  clientId = "CLIENT_ID") 
-```
+### Init
 
 #### Kotlin
 
 ```kotlin
-val authHandler = AuthHandler(configuration) 
+val provider = IdentityProvider(Region.EUCentral1, "<clientId>") 
 ```
 
 #### Typescript
 
 ```typescript
-import {AuthHandlerJS} from '@liftric/auth';
+import {IdentityProviderJS} from '@liftric/auth';
 
-const auth = new AuthHandlerJS('<regionString>', '<clientId>');
+const provider = new IdentityProviderJS('<regionString>', '<clientId>');
 ```
 
-### API
-
-General usage of the request methods.
+### Usage
 
 #### Kotlin
 
-All methods are suspending and will return a `Result<T>` object which wraps the desired return object `T` and can contain an exception.
+All methods are suspending and return a [Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/).
 
 ```kotlin
-val response = signUp(username = "user", password = "password")
-if (response.isSuccess) {
-    println(signUpResponse.getOrNull())
-} else {
-    println(signUpResponse.exceptionOrNull())
-}
+provider.signUp("user", "password").fold(
```

Comment:
That's much more elegant to write! How is that usable within the Swift world, just curious 🤔 !

Reply:
Seems like it doesn't get exported :/

Reply:
I reverted back to the custom Result and added the fold method.